provider/pkg/provider/provider.go

```diff
@@ -1262,6 +1262,12 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 	oldInputs := propMapToUnstructured(olds)
 	newInputs := propMapToUnstructured(news)
 
+	if k.serverSideApplyMode && strings.HasSuffix(urn.Type().String(), "Patch") {
```

Comment:
Nit: would be good to pull `strings.HasSuffix(urn.Type().String(), "Patch")` to a function like `isPatchResource()` or something.

Reply:
Agreed -- I'll follow up on that in a later change.